### PR TITLE
ttl,ddl: disable ttl for fk and temp table

### LINKED ISSUE #39351: TTL Table should not be reference as a parent table by a foreign key

### LINKED ISSUE #39364: Forbid creating/altering a temporary table with TTL options

### PR DESCRIPTION
**Pull request by YangKeao**
Signed-off-by: YangKeao <yangkeao@chunibyo.icu>

### What problem does this PR solve?

Issue Number: close #39364, close #39351

Problem Summary:

TTL table cannot be a temporary table and cannot be referenced as the parent of foreign key.

### What is changed and how it works?

Add several checks for alter ttl, alter forign key, and create table.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### CODE REVIEW

File: ddl/ttl.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ package ddl
 
 import (
 	"fmt"
+	"github.com/pingcap/tidb/sessiontxn"
```
Comment (hawkingrei) [inline ddl/ttl.go:19]: It need to sort.

Comment (YangKeao) [reply ddl/ttl.go]: Fixed.

**Review by hawkingrei — APPROVED** [review]

**Review by xhebox — APPROVED** [review]

**Pull request merged**